classes/statistics/MetricsDAO.inc.php

```diff
@@ -94,9 +94,241 @@ protected function getAssocObjectInfo($submissionId, $contextId) {
+					COUNT(CASE WHEN s.status = ' . STATUS_QUEUED . ' AND s.stage_id = ' . WORKFLOW_STAGE_ID_PRODUCTION . ' THEN 0 END) AS active_production,
+				
+					-- all submissions
+					COUNT(0) AS submission_received,
+					COUNT(CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_ACCEPT . ' THEN 0 END) AS submission_accepted,
+					COUNT(CASE WHEN s.status = ' . STATUS_PUBLISHED . ' AND EXISTS (
+						SELECT 0
+						FROM published_submissions ps
+						WHERE ps.submission_id = s.submission_id
+					) THEN 0 END) AS submission_published,
+					COUNT(CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_INITIAL_DECLINE . ' THEN 0 END) AS submission_declined_initial,
+					COUNT(CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_DECLINE . ' THEN 0 END) AS submission_declined,
+					COUNT(
+						CASE WHEN s.status = ' . STATUS_DECLINED . '
+							AND (ed.decision NOT IN (' . SUBMISSION_EDITOR_DECISION_DECLINE . ', ' . SUBMISSION_EDITOR_DECISION_INITIAL_DECLINE . ')
+								OR ed.decision IS NULL)
+							THEN 0
+						END
+					) AS submission_declined_other,
+					COUNT(CASE WHEN s.status = ' . STATUS_DECLINED . ' THEN 0 END) AS submission_declined_total,
+				
+					-- average days to decide
+					AVG(
+						CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_ACCEPT . '
+							THEN DATEDIFF(ed.date_decided, COALESCE(s.date_submitted, s.last_modified))
+						END
+					) AS submission_days_to_accept,
+					AVG(
+						CASE WHEN ed.decision IN (' . SUBMISSION_EDITOR_DECISION_DECLINE . ', ' . SUBMISSION_EDITOR_DECISION_INITIAL_DECLINE . ')
+							THEN DATEDIFF(ed.date_decided, COALESCE(s.date_submitted, s.last_modified))
+						END
+					) AS submission_days_to_reject,
+					AVG(DATEDIFF(first_ed.date_decided, COALESCE(s.date_submitted, s.last_modified))) AS submission_days_to_first_decide,
+					AVG(DATEDIFF(ed.date_decided, COALESCE(s.date_submitted, s.last_modified))) AS submission_days_to_decide,
+				
+					-- acceptance/rejection rate
+					COUNT(CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_ACCEPT . ' THEN 0 END) / COUNT(0) * 100 AS submission_acceptance_rate,
+					COUNT(CASE WHEN s.status = ' . STATUS_DECLINED . ' THEN 0 END) / COUNT(0) * 100 AS submission_rejection_rate,
+					COUNT(CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_INITIAL_DECLINE . ' THEN 0 END) / COUNT(0) * 100 AS submission_declined_initial_rate,
+					COUNT(CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_DECLINE . ' THEN 0 END) / COUNT(0) * 100 AS submission_declined_rate,
+					COUNT(
+						CASE WHEN s.status = ' . STATUS_DECLINED . '
+							AND (
+								ed.decision NOT IN (' . SUBMISSION_EDITOR_DECISION_DECLINE . ', ' . SUBMISSION_EDITOR_DECISION_INITIAL_DECLINE . ')
+								OR ed.decision IS NULL
+							)
+							THEN 0
+						END
+					) / COUNT(0) * 100 AS submission_declined_other_rate
+				FROM submissions s
+				LEFT JOIN edit_decisions first_ed
+					ON first_ed.edit_decision_id = (
+						SELECT ed.edit_decision_id
```

Comment:
You can optimize this to remove the subselect by adding another LEFT JOIN. Try:
```
LEFT JOIN edit_decisions first_ed ON (first_ed.submission_id = s.submission_id)
LEFT JOIN edit_decisions before_first_d ON (before_first_ed.submission_id = s.submission_id AND before_first_id.edit_decision_id < first_ed.edit_decision_id)
...
WHERE before_first_d.edit_decision_id IS null;
```
(In pseudocode form, think "find an edit decision for which an earlier edit decision doesn't exist -- therefore it'll be the first one.")

That ought to be a lot quicker, I think.

Reply:
@asmecher I avoid using auto-increment IDs to find the first/last record when the order is controlled by another column (in this case a combination of stage_id > round > date_decided, right?). So, I'd rather keep it this way for the sake of rightness.

I checked against the database backup that I have and looks like some data was updated from outside (or maybe an update script), and even though the result will be the same, the rule of `smallest id = first record` was broken:
![image](https://user-images.githubusercontent.com/361921/63653546-de3b3000-c776-11e9-8194-173ff1e21104.png).

Reply:
ps: I can't see much difference here because my `edit_decisions` table has only ~600 records and I'm using an SSD, but the idea is good! I've used it in the past, in another situation, but wouldn't remember to apply it here :D